integration-tests/test/test_genesis_ceremony.py

```diff
@@ -20,22 +28,57 @@
 VALIDATOR_B_KEYPAIR = KeyPair(private_key='1f52d0bce0a92f5c79f2a88aae6d391ddf853e2eb8e688c5aa68002205f92dad', public_key='043c56051a613623cd024976427c073fe9c198ac2b98315a4baff9d333fbb42e')
 
 
-@pytest.mark.xfail
 def test_successful_genesis_ceremony(command_line_options: CommandLineOptions, random_generator: Random, docker_client: DockerClient) -> None:
+    """
+    https://docs.google.com/document/d/1Z5Of7OVVeMGl2Fw054xrwpRmDmKCC-nAoIxtIIHD-Tc/
+    """
     bootstrap_cli_options = {
-        '--required-sigs':  '2',
-        '--duration':       '5min',
-        '--interval':       '10sec',
+        '--deploy-timestamp':   '1',
+        '--required-sigs':      '2',
+        '--duration':           '5min',
+        '--interval':           '10sec',
+    }
+    peers_cli_flags = set(['--genesis-validator'])
+    peers_cli_options = {
+        '--deploy-timestamp':   '1',
+        '--required-sigs':      '2',
     }
     peers_keypairs = [
         VALIDATOR_A_KEYPAIR,
         VALIDATOR_B_KEYPAIR,
     ]
-    with conftest.testing_context(command_line_options, random_generator, docker_client, bootstrap_keypair=CEREMONY_MASTER_KEYPAIR, peers_keypairs=peers_keypairs) as context:
-        with ready_bootstrap(context=context, cli_options=bootstrap_cli_options) as bootstrap:
-            with started_peer(context=context, network=bootstrap.network, bootstrap=bootstrap, name='validator-a', keypair=VALIDATOR_A_KEYPAIR):
-                with started_peer(context=context, network=bootstrap.network, bootstrap=bootstrap, name='validator-b', keypair=VALIDATOR_B_KEYPAIR):
-                    assert False
+    with testing_context(command_line_options, random_generator, docker_client, bootstrap_keypair=CEREMONY_MASTER_KEYPAIR, peers_keypairs=peers_keypairs) as context:
+        with temporary_wallets_file(context.random_generator, [context.bootstrap_keypair] + context.peers_keypairs) as wallets:
+            with ready_bootstrap(context=context, cli_options=bootstrap_cli_options, wallets_file=wallets) as ceremony_master:
+                with started_peer(context=context, network=ceremony_master.network, bootstrap=ceremony_master, name='validator-a', keypair=VALIDATOR_A_KEYPAIR, wallets_file=wallets, cli_flags=peers_cli_flags, cli_options=peers_cli_options) as validator_a:
+                    with started_peer(context=context, network=ceremony_master.network, bootstrap=ceremony_master, name='validator-b', keypair=VALIDATOR_B_KEYPAIR, wallets_file=wallets, cli_flags=peers_cli_flags, cli_options=peers_cli_options) as validator_b:
+                        wait_for_block_approval(context, ceremony_master)
```

Comment:
*puke*

https://stackoverflow.com/questions/893333/multiple-variables-in-python-with-statement

Reply:
with statement can be replaced with explicit calls to `__enter__()` and `__exit__()` if I'm not mistaken, but I think deep nesting is still more readable and explicit.